packages/grafana-ui/src/components/Switch/Switch.tsx

```diff
@@ -36,9 +36,10 @@ export const Switch = React.forwardRef<HTMLInputElement, Props>(
           }}
           id={switchIdRef.current}
           {...inputProps}
+          aria-label={label ?? 'Toggle switch'}
           ref={ref}
         />
-        <label htmlFor={switchIdRef.current} aria-label={label ?? 'Toggle switch'} />
+        <label htmlFor={switchIdRef.current} />
```

Comment:
This removes the label association with the input as now `label` doesn't have any value that labels it.

Reply:
Hey @Clarity-89 thanks for the quick feedback. 

The label association is still there (`htmlFor` in label with `id` in input). The label under Switch component has no visible value, saying that adding the `aria-label` to the element the user needs to interact with, gives the necessary context for the screen readers.
The issue of screen readers reading 2 times the label for this case: `<InlineSwitch showLabel label={`Visible label`} id={'1'} />` is solved.

I indeed noticed that for the InlineSwitch once `showLabel` is passed as props, it will generate 2 labels. If the goal is to re-structure the component I can help, but the change might cause a break change. Please, let me know

Reply:
> The label association is still there (htmlFor in label with id in input).

It is still there indeed, but after you remove the aria-label from the `label` it has nothing that labels it, so the association doesn't work because the label doesn't have any textual value. You can see it also breaks our accessibility check in the CI pipeline.

After a quick look at the `InlineSwitch` component, I don't see an easy way to fix this issue, maybe not rendering the label inside `Switch` if the `showLabel` for the `InlineSwitch` provided, could work 🤔  This would require adding a condition for rendering the label inside the `Switch` component, which should be true by default.